src/Pattern/CaptureCache.php

```diff
@@ -325,12 +325,6 @@ protected function createDirectoryStructure($pathname)
                 $err = ErrorHandler::stop();
                 throw new Exception\RuntimeException("mkdir('{$pathname}', 0{$oct}, true) failed", 0, $err);
             }
-
-            if ($perm !== false && ! chmod($pathname, $perm)) {
```

Comment:
I've spotted this one some time ago already (also with psalm AFAIR).

This check is within an if condition `if ($perm === false)`, so `$perm !== false` is never true here.